File: Day 07/Buy_Sell_Stock_I.cpp
// Multiple Transactions Allowed
#include <iostream>
#include <vector>
using namespace std;

class Solution {
  public:
    int maximumProfit(vector<int> &prices) {
        int profit = 0;
        for(int i = 1; i < prices.size(); i++){
            if(prices[i] > prices[i-1]){
                profit += prices[i] - prices[i-1];
            }
        }
        return profit;
        
    }
};

int main() {
    Solution sol;
    vector<int> prices = {1, 5, 3, 8, 4, 9};
    cout << sol.maximumProfit(prices) << endl; // Output: 14
    return 0;
}

File: Day 07/Buy_Sell_Stock_I.py
# Multiple Transactions Allowed

class Solution:
    def maximum_profit(self, prices):
        profit = 0
        for i in range(1, len(prices)):
            if prices[i] > prices[i-1]:
                profit += prices[i] - prices[i-1]
        return profit

if __name__ == "__main__":
    sol = Solution()
    prices = [1, 5, 3, 8, 4, 9]
    print(sol.maximum_profit(prices))  # Output: 14